File: mobile/library/kotlin/src/io/envoyproxy/envoymobile/UpstreamHttpProtocol.kt
package io.envoyproxy.envoymobile

import java.lang.IllegalArgumentException

/**
 * Available upstream HTTP protocols.
 */
enum class UpstreamHttpProtocol(internal val stringValue: String) {
  HTTP1("http1"),
  HTTP2("http2");

  companion object {
    internal fun enumValue(stringRepresentation: String): UpstreamHttpProtocol {
      return when (stringRepresentation) {
        "http1" -> UpstreamHttpProtocol.HTTP1
        "http2" -> UpstreamHttpProtocol.HTTP2
        else -> throw IllegalArgumentException("invalid value $stringRepresentation")
      }
    }
  }
}

File: mobile/library/swift/src/UpstreamHttpProtocol.swift
import Foundation

/// Available upstream HTTP protocols.
@objc
public enum UpstreamHttpProtocol: Int, CaseIterable {
  case http1
  case http2

  /// String representation of the protocol.
  var stringValue: String {
    switch self {
    case .http1:
      return "http1"
    case .http2:
      return "http2"
    }
  }

  /// Initialize the protocol using a string value.
  ///
  /// - parameter stringValue: Case-insensitive string value to use for initialization.
  init?(stringValue: String) {
    switch stringValue.lowercased() {
    case "http1":
      self = .http1
    case "http2":
      self = .http2
    default:
      return nil
    }
  }
}
